codec-http/src/test/java/io/netty/handler/codec/http/HttpContentCompressorTest.java

```diff
@@ -198,14 +198,54 @@ public void testChunkedContentWithTrailingHeader() throws Exception {
     }
 
     @Test
-    public void testFullContent() throws Exception {
+    public void testFullContentWithContentLength() throws Exception {
         EmbeddedChannel ch = new EmbeddedChannel(new HttpContentCompressor());
         ch.writeInbound(newRequest());
 
-        FullHttpResponse res = new DefaultFullHttpResponse(
+        FullHttpResponse fullRes = new DefaultFullHttpResponse(
                 HttpVersion.HTTP_1_1, HttpResponseStatus.OK,
                 Unpooled.copiedBuffer("Hello, World", CharsetUtil.US_ASCII));
-        res.headers().set(HttpHeaderNames.CONTENT_LENGTH, res.content().readableBytes());
+        fullRes.headers().set(HttpHeaderNames.CONTENT_LENGTH, fullRes.content().readableBytes());
+        ch.writeOutbound(fullRes);
+
+        Object o = ch.readOutbound();
```

Comment:
consider just directly casting to the object you expect like the other tests ... a class cast exception will be thrown if the expected type is not read.